velox/exec/HashTable.cpp

```diff
@@ -1712,6 +1712,11 @@ void HashTable<ignoreNullKeys>::prepareJoinTable(
     otherTables_.emplace_back(std::unique_ptr<HashTable<ignoreNullKeys>>(
         dynamic_cast<HashTable<ignoreNullKeys>*>(table.release())));
   }
+  for (auto& other : otherTables_) {
+    for (int i = 0; i < rows_->columnTypes().size(); ++i) {
+      rows_->updateColumnHasNulls(i, other->rows()->columnHasNulls(i));
```

Comment:
This looks like a hack. I assume there are a number of independent RowContainers each with its own set of null flags. Why are we combining these flags?

Reply:
@zhli1142015 Thank you for clarifying.

> Only the main table is updated in the HashProbe operation and used for extractColumn. 

Here we have a few independent RowContainers and we modify the flag in one of them to reflect the context of all of them. That feels hacky because the flag no longer represents the contents of the RowContainer. I wonder if there is a cleaner way. Can we store this combined flag in the HashTable directly and use in extractColumn?

Reply:
> Can we store this combined flag in the HashTable directly and use in extractColumn?

Updated, thanks.

> I wonder if there is a cleaner way.

How about we only maintain the null flags in the upper layers of the row container, such as the HashTable and Sort Buffer, and then pass the null flags to the extractColumn function?

Reply:
> How about we only maintain the null flags in the upper layers of the row container, such as the HashTable and Sort Buffer, and then pass the null flags to the extractColumn function?

That might be a reasonable option. We can also limit this to cases where we need to maintain the combined flag across containers.

Reply:
I double checked, looks we can't limit above to the case we need to maintian the combined flags across containers. 

I updated the logic according to your suggestion, please help check, thanks.
> Can we store this combined flag in the HashTable directly and use in extractColumn?